com.unity.perception/Tests/Runtime/GroundTruthTests/DatasetCaptureSensorSchedulingTests.cs

```diff
@@ -61,40 +61,95 @@ public void TearDown()
         public IEnumerator SequenceTimeOfNextCapture_ReportsCorrectTime()
         {
             var ego = DatasetCapture.RegisterEgo("ego");
-            var firstCaptureTime = 1.5f;
-            var period = .4f;
-            var sensorHandle = DatasetCapture.RegisterSensor(ego, "cam", "", period, firstCaptureTime);
+            var firstCaptureFrame = 2f;
+            var simulationDeltaTime = .4f;
+            var sensorHandle = DatasetCapture.RegisterSensor(ego, "cam", "", firstCaptureFrame, PerceptionCamera.CaptureTriggerMode.Scheduled, simulationDeltaTime, 0);
 
+            var startTime = firstCaptureFrame * simulationDeltaTime;
             float[] sequenceTimesExpected =
             {
-                firstCaptureTime,
-                period + firstCaptureTime,
-                period * 2 + firstCaptureTime,
-                period * 3 + firstCaptureTime
+                startTime,
+                simulationDeltaTime + startTime,
+                simulationDeltaTime * 2 + startTime,
+                simulationDeltaTime * 3 + startTime
             };
-            for (var i = 0; i < sequenceTimesExpected.Length; i++)
+
+            for (var i = 0; i < firstCaptureFrame; i++)
             {
+                //render the non-captured frames before firstCaptureFrame
                 yield return null;
+            }
+            for (var i = 0; i < sequenceTimesExpected.Length; i++)
+            {
                 var sensorData = m_TestHelper.GetSensorData(sensorHandle);
                 var sequenceTimeActual = m_TestHelper.CallSequenceTimeOfNextCapture(sensorData);
                 Assert.AreEqual(sequenceTimesExpected[i], sequenceTimeActual, 0.0001f);
+                yield return null;
             }
         }
 
+        [UnityTest]
+        public IEnumerator SequenceTimeOfNextCapture_WithInBetweenFrames_ReportsCorrectTime()
```

Comment:
Should add some tests around CaptureTriggerMode.Manual. Not sure if we would want tests with one scheduled and one manual as well as just one manual.

Reply:
I added a couple of scheduling tests for triggering the manual capture and checking if the capture time has been correctly set.

Reply:
I also modified the dual sensor test and added one manual sensor into the mix. This tests for delta times and capture flags being correctly set when coordinating multiple sensors and some of the are manual.